### ISSUE TITLE
Informational messages in the editor

### ISSUE THREAD

**Issue by hanche** [elves/elvish#1437]
Here is a modest feature request, or should I say feature _suggestion_, not to appear to demanding? This was inspired by discussion on #1432.

I suggest a new function `edit:informational`. It should take a single argument, a text or a styled text. This text should be displayed while the editor is active, in the same area where compilation errors are now displayed. If the editor needs to display such an error, it would replace the informational text. But once the error clears, the informational text should be restored.

Running `edit:informational` would replace any informational text already there. And running it with a `$nil` argument would clear it. Finally, the informational text should be automatically cleared at the beginning (or end?) of an edit cycle. (Alternatively, the informational messages could form a stack, with only the top one being displayed. But that seems like over-engineering to me.)

The intended usage of this is to allow the user to install code that will display brief usage information on the function call currently being edited. An example of this in action can be seen in [SLIME: The Superior Lisp Interaction Mode for Emacs](https://common-lisp.net/project/slime/): There, the argument list for the current function call being edited is displayed in the minibuffer. I find it invaluable, and would like to have it in elvish too.

**Comment by hanche** [comment]
Whoops, I just discovered #551. There seems to be substantial overlap. Perhaps I should close this, and turn it into a comment there, instead?

**Comment by xiaq** [comment]
Yes I think it's essentially the same feature. I added a comments there regarding the scope and terminology.

**Comment by xiaq** [comment]
Seems this can be closed then.
